R/mutate.R

```diff
@@ -174,71 +174,80 @@ mutate.data.frame <- function(.data,
                               .keep = c("all", "used", "unused", "none"),
                               .before = NULL,
                               .after = NULL) {
-  # dplyr front-end
   keep <- arg_match(.keep)
 
   by <- compute_by({{ .by }}, .data, by_arg = ".by", data_arg = ".data")
 
-  before <- enquo(.before)
-  after <- enquo(.after)
-
-  # Execution
   cols <- mutate_cols(.data, dplyr_quosures(...), by)
+
   out <- dplyr_col_modify(.data, cols)
 
-  # Extract input for postprocessing
-  used <- attr(cols, "used")
+  names_original <- names(.data)
 
-  # Compact out `NULL` columns that got removed.
-  # These won't exist in `out`, but we don't want them to look "new".
-  # Note that `dplyr_col_modify()` makes it impossible to `NULL` a group column,
-  # which we rely on below.
-  cols <- compact_null(cols)
+  out <- mutate_relocate(
+    out = out,
+    before = {{ .before }},
+    after = {{ .after }},
+    names_original = names_original
```

Comment:
Should this perhaps be

```suggestion
    names_original = names(.data)
```

with removing the extracted variable above?

Reply:
I think I prefer having the local variable there because I think it is a little easier to read. In general I like collecting the variables up front right before calling the function